### PR TITLE
Deprecate `split-by` command

### PR DESCRIPTION
**Pull request by IanManske**
# Description
I'm not quite sure what the point of the `split-by` command is. The only example for the command seems to suggest it's an additional grouping command. I.e., a record that seems to be the output of the `group-by` command is passed to `split-by` which then adds an additional layer of grouping based on a different column.

# User-Facing Changes
Breaking change, deprecated the command.

### CODE REVIEW

**Comment by fdncred** [comment]
It's probably niche but it's one of @andrasio's commands that I'm guessing he used in his covid analysis long ago. Out of respect, I'd rather get his opinion, if he has any, before doing anything to it.

I'm not really sure how to use it on anything but a record of tables.

**Comment by fdncred** [comment]
Ah, this should be an example too, showing that you need to group-by first.
```nushell
❯ [[first_name, last_name, rusty_at, type];
   [Andrés, Robalino, "10/11/2013", A],
   [JT, Turner, "10/12/2013", B],
   [Yehuda, Katz, "10/11/2013", A]] | group-by rusty_at | split-by type | table -e
╭───┬─────────────────────────────────────────────────────────────────────╮
│   │ ╭────────────┬────────────────────────────────────────────────────╮ │
│ A │ │            │ ╭─#─┬─first_name─┬─last_name─┬──rusty_at──┬─type─╮ │ │
│   │ │ 10/11/2013 │ │ 0 │ Andrés     │ Robalino  │ 10/11/2013 │ A    │ │ │
│   │ │            │ │ 1 │ Yehuda     │ Katz      │ 10/11/2013 │ A    │ │ │
│   │ │            │ ╰───┴────────────┴───────────┴────────────┴──────╯ │ │
│   │ ╰────────────┴────────────────────────────────────────────────────╯ │
│   │ ╭────────────┬────────────────────────────────────────────────────╮ │
│ B │ │            │ ╭─#─┬─first_name─┬─last_name─┬──rusty_at──┬─type─╮ │ │
│   │ │ 10/12/2013 │ │ 0 │ JT         │ Turner    │ 10/12/2013 │ B    │ │ │
│   │ │            │ ╰───┴────────────┴───────────┴────────────┴──────╯ │ │
│   │ ╰────────────┴────────────────────────────────────────────────────╯ │
╰───┴─────────────────────────────────────────────────────────────────────╯
```

**Comment by fdncred** [comment]
I kind of like how this groups and splits by two different things.
```nushell
ls | update modified {format date %D} | group-by type | split-by modified | table -e                               10  06:46:07 AM
╭──────────┬────────────────────────────────────────────────────────────────────╮
│          │ ╭──────┬─────────────────────────────────────────────────────────╮ │
│ 09/14/24 │ │      │ ╭─#─┬────────name─────────┬─type─┬──size───┬─modified─╮ │ │
│          │ │ file │ │ 0 │ CITATION.cff        │ file │   812 B │ 09/14/24 │ │ │
│          │ │      │ │ 1 │ CODE_OF_CONDUCT.md  │ file │  3.4 KB │ 09/14/24 │ │ │
│          │ │      │ │ 2 │ CONTRIBUTING.md     │ file │ 11.2 KB │ 09/14/24 │ │ │
│          │ │      │ │ 3 │ Cross.toml          │ file │   666 B │ 09/14/24 │ │ │
│          │ │      │ │ 4 │ LICENSE             │ file │  1.1 KB │ 09/14/24 │ │ │
│          │ │      │ │ 5 │ README.md           │ file │ 12.3 KB │ 09/14/24 │ │ │
│          │ │      │ │ 6 │ SECURITY.md         │ file │  2.7 KB │ 09/14/24 │ │ │
│          │ │      │ │ 7 │ rust-toolchain.toml │ file │  1.1 KB │ 09/14/24 │ │ │
│          │ │      │ │ 8 │ typos.toml          │ file │   513 B │ 09/14/24 │ │ │
│          │ │      │ ╰─#─┴────────name─────────┴─type─┴──size───┴─modified─╯ │ │
│          │ │      │ ╭─#─┬──name───┬─type─┬──size──┬─modified─╮              │ │
│          │ │ dir  │ │ 0 │ assets  │ dir  │  160 B │ 09/14/24 │              │ │
│          │ │      │ │ 1 │ crates  │ dir  │ 1.3 KB │ 09/14/24 │              │ │
│          │ │      │ │ 2 │ devdocs │ dir  │  224 B │ 09/14/24 │              │ │
│          │ │      │ │ 3 │ scripts │ dir  │  416 B │ 09/14/24 │              │ │
│          │ │      │ │ 4 │ tests   │ dir  │  544 B │ 09/14/24 │              │ │
│          │ │      │ │ 5 │ wix     │ dir  │  160 B │ 09/14/24 │              │ │
│          │ │      │ ╰─#─┴──name───┴─type─┴──size──┴─modified─╯              │ │
│          │ ╰──────┴─────────────────────────────────────────────────────────╯ │
│          │ ╭──────┬─────────────────────────────────────────────────╮         │
│ 10/06/24 │ │      │ ╭─#─┬────name────┬─type─┬───size───┬─modified─╮ │         │
│          │ │ file │ │ 0 │ Cargo.lock │ file │ 179.6 KB │ 10/06/24 │ │         │
│          │ │      │ │ 1 │ Cargo.toml │ file │   9.2 KB │ 10/06/24 │ │         │
│          │ │      │ ╰───┴────────────┴──────┴──────────┴──────────╯ │         │
│          │ ╰──────┴─────────────────────────────────────────────────╯         │
│          │ ╭──────┬────────────────────────────────────────────────╮          │
│ 10/02/24 │ │      │ ╭─#─┬────name────┬─type─┬──size───┬─modified─╮ │          │
│          │ │ file │ │ 0 │ toolkit.nu │ file │ 19.7 KB │ 10/02/24 │ │          │
│          │ │      │ ╰───┴────────────┴──────┴─────────┴──────────╯ │          │
│          │ ╰──────┴────────────────────────────────────────────────╯          │
│          │ ╭─────┬───────────────────────────────────────────╮                │
│ 09/15/24 │ │     │ ╭─#─┬──name───┬─type─┬─size──┬─modified─╮ │                │
│          │ │ dir │ │ 0 │ benches │ dir  │ 128 B │ 09/15/24 │ │                │
│          │ │     │ ╰───┴─────────┴──────┴───────┴──────────╯ │                │
│          │ ╰─────┴───────────────────────────────────────────╯                │
│          │ ╭─────┬──────────────────────────────────────────╮                 │
│ 10/04/24 │ │     │ ╭─#─┬──name──┬─type─┬─size──┬─modified─╮ │                 │
│          │ │ dir │ │ 0 │ docker │ dir  │  96 B │ 10/04/24 │ │                 │
│          │ │     │ │ 1 │ src    │ dir  │ 384 B │ 10/04/24 │ │                 │
│          │ │     │ ╰───┴────────┴──────┴───────┴──────────╯ │                 │
│          │ ╰─────┴──────────────────────────────────────────╯                 │
│          │ ╭─────┬──────────────────────────────────────────╮                 │
│ 10/03/24 │ │     │ ╭─#─┬──name──┬─type─┬─size──┬─modified─╮ │                 │
│          │ │ dir │ │ 0 │ target │ dir  │ 224 B │ 10/03/24 │ │                 │
│          │ │     │ ╰───┴────────┴──────┴───────┴──────────╯ │                 │
│          │ ╰─────┴──────────────────────────────────────────╯                 │
╰──────────┴────────────────────────────────────────────────────────────────────╯
```

**Comment by fdncred** [comment]
I don't want to lose this command. Let's just keep it.

**Comment by IanManske** [comment]
Hmm, I still don't see the use case.

- it only works on group-by output
- they cannot be chained together
- it's essentially another group-by command and probably does not accomolish anything group-by cannot already
- anecdoately, I have seen no one use the command

**Comment by hustcer** [comment]
We can add a warning message for this command in the following release to tell the user this command will be removed in the future, and remove it after a few releases if no one objects

**Comment by IanManske** [comment]
> We can add a warning message for this command in the following release to tell the user this command will be removed in the future, and remove it after a few releases if there aren't too many objections

That works for me :+1:

**Comment by NotTheDr01ds** [comment]
Recreated as a custom-command in case anyone misses it:

```nu
def split-by [ splitter: string ]: record -> record {
  transpose -d keys values
  | flatten --all
  | group-by --to-table ([$splitter] | into cell-path)
  | update items {
      group-by --to-table keys
      | update items { reject keys }
      | transpose -dr
    }
  | transpose -dr
}
```

**Comment by Bahex** [comment]
It's probably safe to deprecate `split-by` now, since `group-by` completely covers its use case.

#14337
> Grouping by multiple columns, without `--to-table` returns a nested structure. This is equivalent to `$data | group-by year | split-by lang`, making `split-by` obsolete.
> 
> ```nushell
> > $data | group-by lang year
> ╭────┬─────────────────────────────────────────╮
> │    │ ╭──────┬──────────────────────────────╮ │
> │ rb │ │      │ ╭─#─┬──name──┬─lang─┬─year─╮ │ │
> │    │ │ 2019 │ │ 0 │ andres │ rb   │ 2019 │ │ │
> │    │ │      │ ╰─#─┴──name──┴─lang─┴─year─╯ │ │
> │    │ ╰──────┴──────────────────────────────╯ │
> │    │ ╭──────┬─────────────────────────────╮  │
> │ rs │ │      │ ╭─#─┬─name─┬─lang─┬─year─╮  │  │
> │    │ │ 2019 │ │ 0 │ jt   │ rs   │ 2019 │  │  │
> │    │ │      │ ╰─#─┴─name─┴─lang─┴─year─╯  │  │
> │    │ │      │ ╭─#─┬─name──┬─lang─┬─year─╮ │  │
> │    │ │ 2021 │ │ 0 │ storm │ rs   │ 2021 │ │  │
> │    │ │      │ ╰─#─┴─name──┴─lang─┴─year─╯ │  │
> │    │ ╰──────┴─────────────────────────────╯  │
> ╰────┴─────────────────────────────────────────╯
> ```

**Comment by fdncred** [comment]
I've tested the command above and changed it slightly to make it work.
```nushell
ls | update modified {format date %D} | group-by modified type | table -e
```
I'm fine with deprecating `split-by` now. Thanks!

File: crates/nu-protocol/src/errors/shell_error.rs [hunk 1]
```diff
@@ -1330,14 +1330,15 @@ This is an internal Nushell error, please file an issue https://github.com/nushe
         span: Span,
     },
 
-    #[error("Deprecated: {old_command}")]
-    #[diagnostic(help("for more info see {url}"))]
```
Comment (sholderbach) [inline crates/nu-protocol/src/errors/shell_error.rs:1334]: Did we ever satisfy that with useful URLs :P ? 
Slightly worried with those half sentence format strings, when folks may treat that as free text and never check the format string but deprecation warnings shouldn't live too long.

**Review by sholderbach — APPROVED** [review]

**Pull request merged**